src/Corax/Queries/BinaryMatch.Boolean.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Corax.Queries
+{
+    unsafe partial struct BinaryMatch<TInner, TOuter>
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static BinaryMatch<TInner, TOuter> YieldAnd(in TInner inner, in TOuter outer)
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            static int FillFunc(ref BinaryMatch<TInner, TOuter> match, Span<long> matches)
+            {
+                ref var inner = ref match._inner;
+                ref var outer = ref match._outer;
+
+                while (true)
+                {
+                    var results = inner.Fill(matches);
+                    if (results == 0)
+                        return 0;
+
+                    results = outer.AndWith(matches.Slice(0, results));
+                    if (results != 0)
+                        return results;
+                }
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
this is called via a function pointer, I don't think it _can_ be inlined

Reply:
For now it can't. Don't know in the near future.